peer/pendingheap/list_test.go

```diff
@@ -142,25 +142,25 @@ func TestPeerHeapList(t *testing.T) {
 			},
 			expectedRunning: true,
 		},
-		{
-			msg: "start many and choose",
-			retainedAvailablePeerIDs: []string{"1", "2", "3", "4", "5", "6"},
-			expectedAvailablePeers:   []string{"1", "2", "3", "4", "5", "6"},
-			peerListActions: []PeerListAction{
-				StartAction{},
-				UpdateAction{AddedPeerIDs: []string{"1", "2", "3", "4", "5", "6"}},
-				ChooseAction{ExpectedPeer: "1"},
-				ChooseAction{ExpectedPeer: "2"},
-				ChooseAction{ExpectedPeer: "3"},
-				ChooseAction{ExpectedPeer: "4"},
-				ChooseAction{ExpectedPeer: "5"},
-				ChooseAction{ExpectedPeer: "6"},
-				ChooseAction{ExpectedPeer: "1"},
-			},
-			expectedRunning: true,
-		},
-		{
-			msg: "assure start is idempotent",
+		// {
+		// 	msg:                      "start many and choose",
+		// 	retainedAvailablePeerIDs: []string{"1", "2", "3", "4", "5", "6"},
+		// 	expectedAvailablePeers:   []string{"1", "2", "3", "4", "5", "6"},
+		// 	peerListActions: []PeerListAction{
+		// 		StartAction{},
+		// 		UpdateAction{AddedPeerIDs: []string{"1", "2", "3", "4", "5", "6"}},
+		// 		ChooseAction{ExpectedPeer: "1"},
+		// 		ChooseAction{ExpectedPeer: "2"},
+		// 		ChooseAction{ExpectedPeer: "3"},
+		// 		ChooseAction{ExpectedPeer: "4"},
+		// 		ChooseAction{ExpectedPeer: "5"},
+		// 		ChooseAction{ExpectedPeer: "6"},
+		// 		ChooseAction{ExpectedPeer: "1"},
+		// 	},
+		// 	expectedRunning: true,
+		// },
```

Comment:
Ah, I see. By subverting the random number generator you want to be able to write tests with consecutive order.

Maybe we can thread a test option that just carries a function that returns the next tie breaker instead of an RNG seed.